### PR TITLE
Optional see all usages

### PR DESCRIPTION
**Pull request by attfarhan**
Fixes https://github.com/sourcegraph/issues-uber/issues/218.

- Fixes "see all usages" links because they didn't actually seem to be working at all on sourcegraph.com.
- Turns off see all usages links by default.
- Adds a setting "pyImports.showAllUsagesLinks" to enable see all usages links.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -33,11 +34,17 @@
...
     "parcel-bundler": "^1.10.3",
-    "sourcegraph": "^18.4.0",
+    "rxjs": "^6.5.3",
+    "sourcegraph": "23.0.1",
     "tslint": "^5.11.0",
-    "typescript": "^3.1.4"
+    "typescript": "^3.7.3"
+  },
+  "dependencies": {
+    "@sourcegraph/prettierrc": "^3.0.1"
```
Comment (lguychard) [inline package.json:48]: This should be in devDepencies

File: src/python-imports-search.ts [hunk 2]
```diff
@@ -1,67 +1,91 @@
...
-               const pyPkg = pyImportsFilter && pyImportsFilter.length >= 1 ? pyImportsFilter[1] : ''
-               const singleImport = '^import\\s' + pyPkg + '$'
-               const namedImport = '^import\\s' + pyPkg + '\\sas\\s[^\\s]*'
-               const scopedImport = '^from\\s(.*)\\simport\\s(.*)' + pyPkg
-               return query.replace(pyImportsRegex  , `(${singleImport}|${namedImport}|${scopedImport}) lang:python `)
-           }
-           return query
-        }
-   })
+    const subscriptions = new Subscription()
```
Comment (lguychard) [inline src/python-imports-search.ts:10]: Please use [`ExtensionContext`](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@5e9f971/-/blob/packages/sourcegraph-extension-api/src/sourcegraph.d.ts#L1368-1399) (passed to the `activate` function) for this.

File: src/python-imports-search.ts [hunk 3]
```diff
@@ -1,67 +1,91 @@
...
+                  from(activeWindow.activeViewComponentChanges).pipe(map(() => activeWindow.visibleViewComponents))
+              )
+          )
+        : from(sourcegraph.workspace.openedTextDocuments).pipe(
+              map(() => (sourcegraph.app.activeWindow && sourcegraph.app.activeWindow.visibleViewComponents) || [])
+          )
+
+    subscriptions.add(
+        editorsChanges.subscribe(codeEditors => {
+            const codeEditor = codeEditors[0]
```
Comment (lguychard) [inline src/python-imports-search.ts:48]: Wouldn't you want to decorate _all_ editors when editorsChanges emits?

File: src/python-imports-search.ts (hunk 2 above)
Comment (attfarhan) [reply src/python-imports-search.ts:10]: https://github.com/sourcegraph/python-imports-search/commit/232b7aaa84eff52dc9f4ea4396b49724da0656f4

File: src/python-imports-search.ts (hunk 3 above)
Comment (attfarhan) [reply src/python-imports-search.ts:48]: Is this to accommodate diffs/PRs?

Comment (lguychard) [reply src/python-imports-search.ts:48]: Yes.

File: src/python-imports-search.ts (hunk 2 above)
Comment (lguychard) [reply src/python-imports-search.ts:10]: Huh. I did that thing where I clicked the GitHub email review request link and it took me to a subset of the changes. Sorry!

**Pull request merged**